src/Sulu/Bundle/AdminBundle/Resources/js/containers/List/List.js

```diff
@@ -241,37 +246,73 @@ class List extends React.Component<Props> {
         return deletePromise;
     };
 
-    @action handleDeleteResponseError = (response: Object) => {
-        if (response.status !== 409) {
-            throw response;
-        }
-
+    @action closeAllDialogs = () => {
         this.showDeleteDialog = false;
         this.showDeleteSelectionDialog = false;
-        this.showDeleteLinkedDialog = true;
+        this.referencingResourcesData = undefined;
+        this.dependantResourcesData = undefined;
+    };
+
+    @action handleDeleteResponseError = (response: Object) => {
+        const {onDeleteError} = this.props;
+
         response.json().then(action((data) => {
-            this.referencingItemsForDelete.splice(0, this.referencingItemsForDelete.length);
-            this.referencingItemsForDelete.push(...data.items);
+            this.closeAllDialogs();
 
-            const deleteLinkedPromise: Promise<ResolveDeleteArgument> = new Promise(
-                (resolve) => this.resolveDelete = resolve
-            );
+            if (response.status === 409 && data.code === ERROR_CODES.REFERENCING_RESOURCES_FOUND) {
+                this.referencingResourcesData = {
+                    resource: data.resource,
+                    referencingResources: data.referencingResources,
+                    referencingResourcesCount: data.referencingResourcesCount,
+                };
 
-            deleteLinkedPromise.then(action((response) => {
-                if (!response.deleted) {
-                    this.showDeleteDialog = false;
-                    this.showDeleteSelectionDialog = false;
-                    this.showDeleteLinkedDialog = false;
-                    return response;
-                }
+                const promise: Promise<ResolveDeleteArgument> = new Promise(
+                    (resolve) => this.resolveDelete = resolve
```

Comment:
i feel like the whole `this.resolveDelete` thing is quite hard to understand 🙈 
i am not sure, but i think we could remove it if we set `this.idToBeDeleted` instead of `this.showDeleteDialog` and delete the item directly in the `this.handleDeleteDialogConfirmClick` method